### PR TITLE
Check DQM plot dependency on specific BuildFiles

### PR DESCRIPTION
**Pull request by aperloff**
#### PR description:

This PR adds a single line to three separate BuildFIle.xml. Presumably these additional lines should have been included before this, but weren't, possibly because of CMSSW header-only linkings. Theoretically these changes shouldn't have any effect. However, we have seen an effect on the DQM plots in #34550 and we would like to see if these changes, on their own, will have a similar affect

#### PR validation:

The PR was formed by doing:
```bash
cmsrel CMSSW_12_0_X_2021-07-26-1100
cd CMSSW_12_0_X_2021-07-26-1100/src/
cmsenv
git-cms-init
git-cms-addpkg Calibration/EcalAlCaRecoProducers
git-cms-addpkg DQMOffline/EGamma
git-cms-addpkg HLTrigger/Muon
<edit the relevant lines>
git-cms-checkdeps -a
scram b -j 8
```

Everything compiles just fine and no additional dependencies were checked out, as expected.

@kpedro88

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by jfernan2** [comment]
thank you @aperloff !

**Comment by aperloff** [comment]
@jfernan2 I'm not seeing any differences. Can you please confirm?

File: DataFormats/L1TrackTrigger/BuildFile.xml [hunk 1]
```diff
@@ -7,6 +7,7 @@
 <use name="DataFormats/TrackerCommon"/>
 <use name="FWCore/MessageLogger"/>
 <use name="FWCore/Utilities"/>
+<use name="hls"/>
```
Comment (davidlange6) [inline DataFormats/L1TrackTrigger/BuildFile.xml:10]: the need for this is introduced in #34550 - so should be included there rather than in a separate PR

**Comment by kpedro88** [comment]
please test

File: DataFormats/L1TrackTrigger/BuildFile.xml (hunk 1 above)
Comment (aperloff) [reply DataFormats/L1TrackTrigger/BuildFile.xml:10]: Yes, that's what will ultimately happen (we hope). We are seeing some strange changes in the DQM monitoring tests in #34550, so we are using this draft PR to try to track them down. Because there were no changes due to the non-L1TrackTrigger build files we decided to check the inclusion of the HLS library.

**Comment by jfernan2** [comment]
> @jfernan2 I'm not seeing any differences. Can you please confirm?

Yes, no differences. And given the passive answer from the EGamma DQM contact, I will sign your PR once the current test is over.
Thanks a lot!

**Comment by aperloff** [comment]
So it looks like the tests that failed had to do with the same Relval-INPUT test timeout issue from https://github.com/cms-sw/cmssw/issues/34633. That said, the DQM checks don't appear to show anything troubling. I would assume this means that there are no issues with the libraries added to the build files in #34550. @jfernan2 , if you agree with this assessment I suggest we close this PR **without** merging and sign-off/move-ahead with #34550.

**Comment by jfernan2** [comment]
I agree, thanks a lot!
